### ISSUE TITLE
v0.5 crashes in UCI mode

### ISSUE THREAD

**Issue by evalon32** [glinscott/leela-chess#287]
E.g.:
```
$ ./lczero -w weights_120.txt
go infinite
...
src/UCTNode.cpp:334: UCTNode* UCTNode::uct_select_child(Color, bool): Assertion `best != nullptr' failed.
```
Happens consistently in UCI mode (but not in train/bench mode). #274 seems to be the culprit (the crash goes away when I reset to ca859de5ae4dfcd3abe4f1edad045e3d4d6ce7ed)

**Comment by DavidNHill** [comment]
It played okay in Arena, but when I tried analysing a game with vsn 0.5 it crashed using Command line "-w network-116 -t2 --noponder -v16000"

Source
lczero.exe

Summary
Stopped working

Date
‎11/‎04/‎2018 10:20

Status
Report sent

Description
Faulting Application Path:	D:\Chess\Leela chess\lczero.exe

Problem signature
Problem Event Name:	APPCRASH
Application Name:	lczero.exe
Application Version:	0.0.0.0
Application Timestamp:	5acd967c
Fault Module Name:	lczero.exe
Fault Module Version:	0.0.0.0
Fault Module Timestamp:	5acd967c
Exception Code:	c0000005
Exception Offset:	0000000000045920
OS Version:	10.0.16299.2.0.0.768.101
Locale ID:	2057
Additional Information 1:	bbd0
Additional Information 2:	bbd07baf349017e94a803756affece31
Additional Information 3:	b308
Additional Information 4:	b30870ccc42fd8c4eb21437862e29b6c

Extra information about the problem
Bucket ID:	fb49e6ae6710ea1005dc6fce1e6eca14 (1575256900737354260)

**Comment by DavidNHill** [comment]
It can be stopped with "Terminate Engine Process" (right-click the analysis window in Arena).

What I found was with -t1 it ended okay (-v0 and manually terminated) with -t2 it failed (-v0 and manually terminated)

With -v16000 it always failed regardless of -t1 or -t2.

**Comment by Akababa** [comment]
Can anyone test to see if this fixes it? I tried running the above commands and it works on my machine.

https://github.com/Akababa/leela-chess-1/tree/uci-async

**Comment by evalon32** [comment]
@Akababa Fixes it for me, thanks!

**Comment by Akababa** [comment]
@evalon32 is it the exact same error on UCTNode.cpp:334? What does it output after "go infinite"?

**Comment by evalon32** [comment]
@Akababa Yes, just double-checked and it's the exact same error.
```
go infinite
update_root, 0 -> 0 expanded nodes (0.0% reused)
NNCache: 0/0 hits/lookups = 0.0% hitrate, 0 inserts, 0 size
RNG seed: 0xd360537f0fe90cb6 (thread: 4690728743462890150)
info depth 6 nodes 2 nps 77 score cp -122 winrate 20.71% time 12 pv e4 e6
RNG seed: 0x9a861d64ca956632 (thread: 18227255271703941185)
info depth 7 nodes 4 nps 188 score cp -78 winrate 29.62% time 15 pv e4 e6 c4 c5
info depth 8 nodes 7 nps 353 score cp -51 winrate 36.11% time 16 pv e4 e6 c4 c5 Nf3
info depth 9 nodes 9 nps 421 score cp -41 winrate 38.69% time 18 pv e4 e6 c4 c5 Nf3 Nc6
info depth 10 nodes 15 nps 636 score cp -25 winrate 42.96% time 21 pv e4 e6 d4 d5 e5 c5
info depth 11 nodes 25 nps 923 score cp -15 winrate 45.80% time 25 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6
info depth 12 nodes 43 nps 1200 score cp -7 winrate 47.96% time 34 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6 Be2
info depth 13 nodes 75 nps 1542 score cp -3 winrate 49.17% time 47 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6 Be2 e5 d3 Be7
info depth 14 nodes 132 nps 1819 score cp 0 winrate 50.05% time 71 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6 Be2 e5 d3 Be7 O-O d6
info depth 15 nodes 235 nps 2108 score cp 1 winrate 50.49% time 110 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6 Be2 e5 d3 Be7 O-O d6 h3
info depth 16 nodes 415 nps 2379 score cp 2 winrate 50.73% time 173 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6 Be2 e5 d3 Be7 O-O d6 h3
info depth 17 nodes 728 nps 2615 score cp 3 winrate 51.03% time 277 pv e4 e6 c4 c5 Nf3 Nc6 Nc3 Nf6 Be2 e5 d3 Be7 O-O d6 h3
lczero: UCTNode.cpp:334: UCTNode* UCTNode::uct_select_child(Color, bool): Assertion `best != nullptr' failed.
Aborted (core dumped)
```

**Comment by killerducky** [comment]
Workaround in #291. Possible fix WIP in #293.
